hayeon/programmers-0/최빈값-구하기.js

```diff
@@ -0,0 +1,18 @@
+function solution(array) {
+  const max = Math.max(...array);
+  const count = Array(max + 1).fill(0);
+  array.map((el) => (count[el] += 1));
+
+  const modeCount = Math.max(...count);
+  const mode = count.filter((i) => i === modeCount);
+  return mode.length !== 1 ? -1 : count.findIndex((i) => i === mode[0]);
+
+  return count;
+}
```

Comment:
```suggestion
function solution(array) {
  const max = Math.max(...array);
  const count = Array(max + 1).fill(0);
  array.forEach((el) => (count[el] += 1));

  const modeCount = Math.max(...count);
  const mode = count.filter((el) => el === modeCount);
    
  return mode.length !== 1 ? -1 : count.findIndex((el) => el === mode[0]);
}
```
- 객체가 아닌 배열을 이용한 풀이 잘봤습니다 🙇‍♂️ 단, 아쉬운 점이라면 맨 밑에 `return`문이 2개가 있습니다 맨 마지막 count 반환은 불 필요한 코드입니다.
- 또한 filter, findexIndex에서 콜백 함수의 인자로 `i`를 사용하셨는데 `i`는 보통 인덱스를 의미하는데 여기서는 순회하는 각 배열 오소를 의미하는 변수명이 적합할 것 같습니다. map에서처럼 `el` 또는 `item`이 적절해 보입니다.
- 중간에 array.map 실행하는 코드가 있는데 따로 새로운 배열을 반환할 필요 없이 단순 순회라면 `forEach`가 적절하다고 생각이 드네요 👍
- 다른 코드를 참고하더라도 다시 한번 검토해주시길 바랄게요 ㅎㅎ!

Reply:
헉,,마지막에 count 반환하는거 제가 임의로 작성했다가 안지웠나봐요ㅠㅠ 항상 검토한다고 했는데,, 문제가 어려워서 정신 놨나봅니다.... 다시 한 번 더 복습하겠습니닷!!!